### PR TITLE
Release Notes for 202012 branch

### PR DESCRIPTION
**Pull request by kannankvs**
release notes for 202012 branch

### CODE REVIEW

**Comment by anshuv-mfst** [comment]
Hi Kannan - please remove Chassis infrastructure and Distributed VoQ architecture from the release notes.

**Comment by anshuv-mfst** [comment]
@kannankvs- please make following changes: 
1. Management framework - please remove 5553, 390, 6148 from the list as these are open. 
2. please remove distributed VoQ architecture feature from the release notes.

File: doc/SONiC_202012_Release_Notes.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+#### SONiC entity MIB extensions
+The Entity MIB contains several groups of MIB objects: entityPhysical group, entityLogical group and so on. Currently SONiC only implemented part of the entityPhysical group following RFC2737. Since entityPhysical group is mostly common used, this extension will focus on entityPhysical group and leave other groups for future implementation. The group entityPhysical contains a single table called "entPhysicalTable" to identify the physical components of the system. 
+<br> **Pull Requests** :  [134](https://github.com/Azure/sonic-platform-common/pull/134), [102](https://github.com/Azure/sonic-platform-daemons/pull/102), [5645](https://github.com/Azure/sonic-buildimage/pull/5645), [168](https://github.com/Azure/sonic-snmpagent/pull/168)  & [2379](https://github.com/Azure/sonic-mgmt/pull/2379)
+
+#### FRR BGP NBI
+This feature extends and provides unified configuration and management capability for FRR-BGP features used in SONiC. This allows the user to configure & manage FRR-BGP using SONiC Management Framework with Open Config data models via REST, gNMI and also provides access via SONiC Management Framework CLI as well.
```
Comment (venkatmahalingam) [inline doc/SONiC_202012_Release_Notes.md:86]: We can use this PR change to load any BGP/route-map configurations from config-DB as well independent of mgmt-framework NBI.

File: doc/SONiC_202012_Release_Notes.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+#### Kubernetes (docker to be controlled by Kubernetes)
+This feature deals in depth with kubernetes-support. With this feature, an image could be downloaded from external repositaries and kubernetes does the deployment. The external Kubernetes masters could be used to deploy container image updates at a massive scale, through manifests. This new mode is referred as "kubernetes mode". 
+<br> **Pull Requests** : [5421](https://github.com/Azure/sonic-buildimage/pull/5421), [1133](https://github.com/Azure/sonic-utilities/pull/1133) 
+
+#### Management Framework (Phase 2)
+Management framework makes use of the translation library (Translib) written in golang to convert the data models exposed to the management clients into the Redis ABNF schema format. Supported management servers can make use of the Translib to convert the incoming payload to SONiC ABNF schema and vice versa depending on the incoming request. Translib will cater to the needs of REST and gNMI servers. Later the Translib can be enhanced to support other management servers if needed. This framework will support both standard and custom YANG models for communication with the corresponding management servers.
```
Comment (ben-gale) [inline doc/SONiC_202012_Release_Notes.md:99]: Please replace this description with the following: -

"The SONiC Management Framework provides a unified UI stack, enabling SONiC to be managed through an Industry-standard CLI, RESTCONF and gNMI (using OpenConfig data models). It also offers a rapid development environment for these user-interfaces, based upon automatic code generation from YANG models.

This set of enhancements extends the framework, including (but not limited to) the following additions: -
- CLI: Usability enhancements, performance enhancements, User-document generation
- Translib Infrastructure: Much of the UI infrastructure is separated out into a new repo (sonic-mgmt-common), de-coupling it from the UI code (which remains in the sonic-mgmt-framework repo). This allows the sonic-telemetry container (which also uses the infra) to be built and used independently from sonic-mgmt-framework. Other enhancements include support for RPC, YANG versioning, bulk calls (e.g. from gNMI), Authorization, and other code optimizations
- CVL: This is an infrastructure component to perform UI validations. It is enhanced here to support Dynamic Port Breakout configuration dependency and for improved performance.
- Transformer: This is an infrastructure component to perform YANG tree to configuration object translations. Enhancements include support for more overloading functions, cascaded operations (for dynamic port breakout) and aliasing
- REST Server: Enhancements include RBAC, RESTCONF, YANG module discovery, statistics and other optimizations
- Advance to Python 3.7
- Use of DBus for communication into the host
- Build enhancements

File: doc/SONiC_202012_Release_Notes.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+#### Multi-DB enhancement-Part 2
```
Comment (ben-gale) [inline doc/SONiC_202012_Release_Notes.md:112]: Description please?

File: doc/SONiC_202012_Release_Notes.md [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+**Pull Requests** : [1165](https://github.com/Azure/sonic-utilities/pull/1165), [106](https://github.com/Azure/sonic-platform-common/pull/106)
+
+#### PDDF advance to SONiC Platform 2.0, BMC
+SONiC OS is portable across different network devices with supported ASIC via Switch Abstraction Interface (SAI). These devices primarily differ in the way various device specific hardware components are accessed, and thus require custom device drivers and python plugins. Each platform vendor implements these custom device drivers and plugins. The feature requirement is to support a SONiC platform driver development framework to enable rapid development of custom device drivers and plugins.
```
Comment (ben-gale) [inline doc/SONiC_202012_Release_Notes.md:120]: Please say as follows: -

PDDF is a rapid platform development environment for SONiC, enabling new platforms to be quickly integrated into the SONiC eco-system by automating much of the routine development using simple description files. it supports both the SONiC Platform 1.0 and 2.0 designs (though 2.0 is preferred). Also supports BMC.

File: doc/SONiC_202012_Release_Notes.md [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+<br> **Pull Requests** : [4756](https://github.com/Azure/sonic-buildimage/pull/4756), [940](https://github.com/Azure/sonic-utilities/pull/940), [92](https://github.com/Azure/sonic-platform-common/pull/92), [3387](https://github.com/Azure/sonic-buildimage/pull/3387), [624](https://github.com/Azure/sonic-utilities/pull/624), [62](https://github.com/Azure/sonic-platform-common/pull/62) 
+
+#### Support hardware reboot/reload reason (Streaming Telemetry)
+This feature enables sonic streaming telemetry agent to send Reboot-cause information. During the boot, the determine-reboot-cause service ( previously process-reboot-cause) determines the last reboot-cause based on the hardware reboot-cause and the software reboot-cause information and determine-reboot-cause service will save the JSON-formatted last previous reboot cause information to "/host/reboot-cause/history/" with adding timestamp at the end of file name.
```
Comment (ben-gale) [inline doc/SONiC_202012_Release_Notes.md:124]: SONiC not sonic (global comment).

You might want to add a little punctuation to the second sentence.

File: doc/SONiC_202012_Release_Notes.md [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+<br> **Pull Requests** : [5562](https://github.com/Azure/sonic-buildimage/pull/5562), [1154](https://github.com/Azure/sonic-utilities/pull/1154) 
+
+#### System health and system LED
+System health monitor is intended to monitor both critical services and peripheral device status and leverage system log, system status LED to and CLI command output to indicate the system status.In current SONiC implementation, we already have Monit which is monitoring the critical services status and also have a set of daemons.System health monitoring service will not monitor the critical services or devices directly, it will reuse the result of Monit and PMON daemons to summary the current status and decide the color of the system health LED.
```
Comment (ben-gale) [inline doc/SONiC_202012_Release_Notes.md:128]: "summary" should be "summarize".

**Review by ben-gale — CHANGES_REQUESTED** [review]

File: doc/SONiC_202012_Release_Notes.md [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+
+#### Gearbox
+The purpose of this feature is to describe PHY functionality and common interface to manage PHY. PHY support the physical layer functionality.  Which is connector between MAC(SerDes) to physical medium such as optical fiber or copper transceivers.  Necessity of PHY depends on platform/hardware design.  Some platforms may be supported without an PHY(PHY Less) or PHY supports as part of ASIC (Internal PHY) and some cases it might be External PHY. External PHY will be used to serve different purposes like gearbox, retimer, MACSEC  and multi gigabit ethernet phy transceivers etc. 
+The PHY has interfaces to connect/communicate with peripherals such as MII interface, SPI interface, power supply, clock and reset, system side interface, and line side interface.
+<br> **Pull Requests** : [347](https://github.com/Azure/sonic-swss-common/pull/347), [931](https://github.com/Azure/sonic-utilities/pull/931), [1321](https://github.com/Azure/sonic-swss/pull/1321), [624](https://github.com/Azure/sonic-sairedis/pull/624) & [4851](https://github.com/Azure/sonic-buildimage/pull/4851).
+
+#### Kubernetes (docker to be controlled by Kubernetes)
```
[the remark this answers is not in the document]
Comment (kannankvs) [reply doc/SONiC_202012_Release_Notes.md:104]: @renukamanavalan  Can you pls comment on this

File: doc/SONiC_202012_Release_Notes.md (hunk 1 above)
Comment (kannankvs) [reply doc/SONiC_202012_Release_Notes.md]: Addressed

File: doc/SONiC_202012_Release_Notes.md [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+
+1. Kernel upgraded from 4.9.110-3deb9u6 (SONiC Release 201904) to 4.9.168-1+deb9u5 for SONiC release. 
+   Change log: https://tracker.debian.org/media/packages/l/linux/changelog-4.9.168-1deb9u5
+2. Docker upgraded from 18.09.2\~3-0\~debian-stretch to 18.09.8\~3-0\~debian-stretch. 
+   Change log: https://docs.docker.com/engine/release-notes/#18098 
+
+# Feature List
+
+#### Consistent ECMP support (fine grain ECMP)
+SAI supports a simple ECMP definition SAI_NEXT_HOP_GROUP_TYPE_ECMP, which can grow or shrink in size, and the members do not assume any order. SAI implementation is responsible for managing the order of the members and the overall group size.
```
[the remark this answers is not in the document]
Comment (kannankvs) [reply doc/SONiC_202012_Release_Notes.md]: Addressed

File: doc/SONiC_202012_Release_Notes.md [hunk 9]
```diff
@@ -0,0 +1,143 @@
...
+**Pull Requests** :  [1315](https://github.com/Azure/sonic-swss/pull/1315), [623](https://github.com/Azure/SONiC/pull/623), [1788](https://github.com/Azure/sonic-mgmt/pull/1788), [4985](https://github.com/Azure/sonic-buildimage/pull/4985), [374](https://github.com/Azure/sonic-swss-common/pull/374), [659](https://github.com/Azure/SONiC/pull/659), [1056](https://github.com/Azure/sonic-utilities/pull/1056), [5518](https://github.com/Azure/sonic-buildimage/pull/5518), [5198](https://github.com/Azure/sonic-buildimage/pull/5198), [693](https://github.com/Azure/SONiC/pull/693).
```
[the remark this answers is not in the document]
Comment (kannankvs) [reply doc/SONiC_202012_Release_Notes.md]: Addressed. HLD document and PR's are given separately.

File: doc/SONiC_202012_Release_Notes.md [hunk 10]
```diff
@@ -0,0 +1,143 @@
...
+<br> **Pull Requests** :  [392](https://github.com/Azure/sonic-buildimage/pull/3992), [1036](https://github.com/Azure/sonic-utilities/pull/1036/files), [5223](https://github.com/Azure/sonic-buildimage/pull/5233),[5163](https://github.com/Azure/sonic-buildimage/pull/5163/files), [5108](https://github.com/Azure/sonic-buildimage/pull/5108/files) & [1036](https://github.com/Azure/sonic-utilities/pull/1036/files).
+
+#### CoPP Config/Management
+During SWSS start, the prebuilt copp.json file is loaded as part of start script swssconfig.sh and written to APP DB. CoppOrch then translates it to Host trap/policer tables and programmed to SAI. With this enhancement, the CoPP tables shall be loaded to Config DB instead of APP DB. The default CoPP json file shall be prebuilt to the image and loaded during initialization. Any Config DB entries present shall be configured overwriting the default CoPP tables. This also ensures backward compatibility.
```
[the remark this answers is not in the document]
Comment (kannankvs) [reply doc/SONiC_202012_Release_Notes.md]: Addressed

File: doc/SONiC_202012_Release_Notes.md [hunk 11]
```diff
@@ -0,0 +1,143 @@
...
+
+#### CoPP Config/Management
+During SWSS start, the prebuilt copp.json file is loaded as part of start script swssconfig.sh and written to APP DB. CoppOrch then translates it to Host trap/policer tables and programmed to SAI. With this enhancement, the CoPP tables shall be loaded to Config DB instead of APP DB. The default CoPP json file shall be prebuilt to the image and loaded during initialization. Any Config DB entries present shall be configured overwriting the default CoPP tables. This also ensures backward compatibility. 
+<br> **Pull Requests** : [358](https://github.com/Azure/sonic-swss-common/pull/358), [1333](https://github.com/Azure/sonic-swss/pull/1333), [4861](https://github.com/Azure/sonic-buildimage/pull/4861) & [1004](https://github.com/Azure/sonic-utilities/pull/1004)
+
+#### Console Support for SONiC (Hardware)
```
[the remark this answers is not in the document]
Comment (kannankvs) [reply doc/SONiC_202012_Release_Notes.md:65]: Addressed

File: doc/SONiC_202012_Release_Notes.md (hunk 2 above)
Comment (kannankvs) [reply doc/SONiC_202012_Release_Notes.md]: Addressed

File: doc/SONiC_202012_Release_Notes.md (hunk 3 above)
Comment (kannankvs) [reply doc/SONiC_202012_Release_Notes.md:153]: Addressed

File: doc/SONiC_202012_Release_Notes.md (hunk 4 above)
Comment (kannankvs) [reply doc/SONiC_202012_Release_Notes.md]: Addressed

File: doc/SONiC_202012_Release_Notes.md (hunk 5 above)
Comment (kannankvs) [reply doc/SONiC_202012_Release_Notes.md]: Addressed

File: doc/SONiC_202012_Release_Notes.md (hunk 6 above)
Comment (kannankvs) [reply doc/SONiC_202012_Release_Notes.md]: Addressed

**Pull request merged**